### PR TITLE
refactor: reduce complexity of transform function

### PR DESCRIPTION
**Pull request by hexqi**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **Chores**
  - Updated exclusion list for the `cmetrics` tool, enhancing analysis precision by ignoring additional configuration files and test directories.

### CODE REVIEW

**Review by gene9831 — DISMISSED** [review]
LGTM

File: packages/build/vite-plugin-meta-comments/src/transform.js [hunk 1]
```diff
@@ -33,6 +33,131 @@ import {
...
+        } from '${COMMON_PACKAGE_NAME}'`
+      )()
+    )
+  }
+}
+
+function handleExportDefaultDeclaration(state) {
+  return function (path) {
+    const comment = path.node.leadingComments
+    const lastComment = comment && comment[comment.length - 1].value
```
Comment (chilingling) [inline packages/build/vite-plugin-meta-comments/src/transform.js]: 没有 comment 是不是直接 return 就好了？或者用 optional chaining 来兼容

File: packages/build/vite-plugin-meta-comments/src/transform.js [hunk 2]
```diff
@@ -33,6 +33,131 @@ import {
...
+    })
+  }
+}
+
+function handleExpressionStatement(state) {
+  return function (path) {
+    const { hooksName, varName, hooksIndex } = state
+    const callName = path.node.expression?.callee?.name
+    const hookName = hooksName[callName]
+    if (hookName) {
```
Comment (chilingling) [inline packages/build/vite-plugin-meta-comments/src/transform.js]: 这里是不是可以用 early return ? 避免 if 条件包裹一大串代码

Comment (hexqi) [reply packages/build/vite-plugin-meta-comments/src/transform.js]: resolved

File: packages/build/vite-plugin-meta-comments/src/transform.js (hunk 1 above)
Comment (hexqi) [reply packages/build/vite-plugin-meta-comments/src/transform.js]: resolved

**Review by chilingling — APPROVED** [review]

**Review by wenmine — APPROVED** [review]

**Pull request merged**